File: codes/r/1.Arrays.r
# Array in R

fruits <- list("apple", "mango", "orange")

cat(paste(sep = "", "fruits, length: ", length(fruits), "\n"))
# fruits, length: 3

for (arrayItemIndex in seq_along(fruits)) {
    arrayItem <- fruits[[arrayItemIndex]]
    cat(paste(sep="", "fruits, for loop, index: ", arrayItemIndex, ", value: ", arrayItem, "\n"))
}
# fruits, foreach loop, index: 1, value: apple
# fruits, foreach loop, index: 2, value: mango
# fruits, foreach loop, index: 3, value: orange

# Array of Objects in R

products <- list(
    list(
        id = "P1",
        name = "bubble gum"
    ),
    list(
        id = "P2",
        name = "potato chips"
    )
)

cat(paste(sep = "", "products, length: ", length(products), "\n"))
# products, length: 2

for (arrayItemIndex in seq_along(products)) {
    arrayItem <- products[[arrayItemIndex]]
    for (objectKey in names(arrayItem)) {
        objectValue <- arrayItem[[objectKey]]
        cat(paste(sep="", "products, for loop, index: ", arrayItemIndex, ", key: ", objectKey, ", value: ", objectValue, "\n"))
    }
}
# products, for loop, index: 1, key: id, value: P1
# products, for loop, index: 1, key: name, value: bubble gum
# products, for loop, index: 2, key: id, value: P2
# products, for loop, index: 2, key: name, value: potato chips

File: codes/python/1.Arrays.py
# Array in Python

fruits = ["apple", "mango", "orange"]

print(f"fruits, length: {len(fruits)}")
# fruits, length: 3


for (array_item_index, array_item) in enumerate(fruits):
    print(f"fruits, for loop, index: {array_item_index}, value: {array_item}")


# fruits, for loop, index: 0, value: apple
# fruits, for loop, index: 1, value: mango
# fruits, for loop, index: 2, value: orange

# Array of Objects in Python

products = [
    {
        "id": "P1",
        "name": "bubble gum"
    },
    {
        "id": "P2",
        "name": "potato chips"
    }
]

print(f"products, length: {len(products)}")
# products, length: 2

for array_item_index, array_item in enumerate(products):
    for object_key, object_value in array_item.items():
        print(f"products, loop, index: {array_item_index}, key: {object_key}, value: {object_value}")
# products, loop, index: 0, key: id, value: P1
# products, loop, index: 0, key: name, value: bubble gum
# products, loop, index: 1, key: id, value: P2
# products, loop, index: 1, key: name, value: potato chips
